app/src/screens/ProposalDetail.js

```diff
@@ -48,43 +59,104 @@ function ProposalDetail({ proposal, onBack, requestToken }) {
     threshold,
   } = proposal
 
-  const myStakes = stakes.filter(({ entity }) =>
-    addressesEqual(entity, connectedAccount)
+  const myStake = useMemo(
+    () => stakes.find(({ entity }) => addressesEqual(entity, connectedAccount)) || {amount: new BigNumber('0')},
+    [stakes]
   )
-  const didIStaked = myStakes.length > 0 && [...myStakes].pop().tokensStaked > 0
+
+
+  const myStakeAmountFormatted = useMemo(() => {
+    if (!myStake) {
+      return '0'
+    }
+    return formatTokenAmount(myStake.amount, stakeToken.tokenDecimals)
+  }, [myStake])
+
+  const totalStaked = useAccountTotalStaked()
+
+  const nonStakedTokens = useMemo(() => stakeToken.balance.minus(totalStaked).plus(myStake.amount), [])
+
+  const formattedMaxAvailableAmount = useMemo(() => {
+    if (!stakeToken) {
+      return '0'
+    }
+    return formatTokenAmount(nonStakedTokens, stakeToken.tokenDecimals)
+  }, [stakeToken])
+
+  const rounding = useMemo(() => {return Math.min(MAX_INPUT_DECIMAL_BASE, stakeToken.decimals)}, [stakeToken])
```

Comment:
I wouldn't use a `useMemo` here for a basic operation. I think it adds more cost than what  we are optimizing for